### PR TITLE
Feature/connection types

### PR DESCRIPTION
**Pull request by dwlfrth**
TO REVIEW

We can merge it so we will be able to work around it after.
Have some stuff would like to integrate to codebase that depends on that.

Review by commit may be easier.
Pay attention to the following commits as they are modifying the actual PacketFence working codebase.
- da2c00efa00df8ea4d35dfd02c836a0baf3b6592
- 6a7eb9a8bff96d01e78a95f03e15ed0aee5b93f2
- 31544378879adfbc5567ca0049fadce24e791fc6

Note that commit c6462aa5a7bb69864e8e71557b82cdb260408539 is overwritten by 31544378879adfbc5567ca0049fadce24e791fc6

### CODE REVIEW

File: lib/pf/Connection.pm [hunk 1]
```diff
@@ -0,0 +1,160 @@
+package pf::Connection;
+
+use Moose;
+use Log::Log4perl qw(get_logger);
```
Comment (rouzier) [inline lib/pf/Connection.pm]: change to pf::log

**Comment by dwlfrth** [comment]
Anyone else ?

**Comment by julsemaan** [comment]
i'll review and comment once i'm done

**Comment by dwlfrth** [comment]
Merge me.. or at least, review me to then merge

**Pull request merged**